### ISSUE TITLE
goroutine counts keeps increasing

### ISSUE THREAD

**Issue by simonzg** [meterio/meter-pov#28]
Here's a screenshot for m09 running about 15 days
you could see the goroutine count keeps increasing and never declines, and the spikes in the chart coincidence with cpu spikes as well.

![Screenshot 2023-04-14 at 1 10 45 PM](https://user-images.githubusercontent.com/7259860/232145708-9cd3fe22-21f2-47eb-a465-07c3695eeb54.png)

**Comment by simonzg** [comment]
This is a goroutine dump from pprof on m09, seems like the function that generates most goroutines is:

github.com/btcsuite/btcd/rpcclient.(*Client).start
github.com/ethereum/go-ethereum/p2p.(*Peer).run

```
▶ node analyze-goroutines.js
82: created by net/http.(*Server).Serve
2: created by github.com/rjeczalik/notify.newNonrecursiveTree
1: created by main.main
1: created by os/signal.Notify.func1.1
1: created by main.handleExitSignal
5: created by github.com/syndtr/goleveldb/leveldb.newSession
20: created by github.com/syndtr/goleveldb/leveldb.openDB
1: created by github.com/meterio/meter-pov/p2psrv/discv5.ListenUDP
1: created by database/sql.OpenDB
1: created by github.com/meterio/meter-pov/chain.New
25: created by github.com/meterio/meter-pov/co.(*Goes).Go
405: created by github.com/btcsuite/btcd/rpcclient.(*Client).start
2: created by github.com/ethereum/go-ethereum/p2p.(*Server).setupListening
2: created by github.com/ethereum/go-ethereum/p2p.newDialScheduler
1: created by github.com/ethereum/go-ethereum/p2p.(*Server).Start
1: created by github.com/meterio/meter-pov/p2psrv/discv5.newNetwork
1: created by github.com/meterio/meter-pov/p2psrv/discv5.(*nodeDB).ensureExpirer.func1
11: created by github.com/ethereum/go-ethereum/p2p.(*Peer).startProtocols
1: created by github.com/meterio/meter-pov/consensus.(*Pacemaker).Start
11: created by github.com/ethereum/go-ethereum/p2p.(*Server).launchPeer
22: created by github.com/ethereum/go-ethereum/p2p.(*Peer).run
200: created by net/http.(*Transport).dialConn
2: created by github.com/meterio/meter-pov/api/subscriptions.(*Subscriptions).pipe
1: created by github.com/meterio/meter-pov/p2psrv/discv5.(*Network).loop
1: created by net/http.(*connReader).startBackgroundRead```

**Comment by simonzg** [comment]
Seems like this is caused by powpool.RelayFrom, which create a rpcclient everytime it's called, and this might has something to do with many TIME_WAIT tcp connections locally as well. This is fixed in c4796660b23907922e0072fefc0e38de36837dd0
